rest_framework/permissions.py

```diff
@@ -172,7 +173,21 @@ def get_required_object_permissions(self, method, model_cls):
         return [perm % kwargs for perm in self.perms_map[method]]
 
     def has_object_permission(self, request, view, obj):
-        model_cls = view.queryset.model
+        try:
+            queryset = view.get_queryset()
+        except AttributeError:
+            queryset = getattr(view, 'queryset', None)
+        except AssertionError:
```

Comment:
I'm catching it just to re-raise it later, with adhoc error message (few lines below), mainly to be consistent with `DjangoModelPermissions`.

I have no strong opinion about it, I'm fine with leaving the original `AssertionError`.
Just tell me what you prefer.